pkg/proofs/proofs.go

```diff
@@ -94,18 +95,25 @@ func CombineResults(results []PrefixSearchResult, steps []BinaryLadderStep) (com
 
 	sortedSteps := make([]BinaryLadderStep, 0, len(results))
 	copy(sortedSteps, steps[:len(results)])
-	slices.SortFunc(sortedSteps, func(a, b BinaryLadderStep) int {
-		hashA := crypto.VRF_proof_to_hash(a.Proof)
-		hashB := crypto.VRF_proof_to_hash(b.Proof)
-		return bytes.Compare(hashA[:], hashB[:])
-	})
+	sortBinaryLadderSteps(sortedSteps)
```

Comment:
Why replace with a function?

Reply:
It's in a trusted function, so Gobra ignores it as we have not axiomatized `slices.SortFunc` yet. In addition, this saves us from reasoning about the closure that defines the sorting criterion.